pkg/handler/employee/employee.go

```diff
@@ -81,6 +83,107 @@ func (h *handler) One(c *gin.Context) {
 }
 
+// Update godoc
+// @Summary Edit employee by id
+// @Description Edit employee by id
+// @Tags Employee
+// @Accept  json
+// @Produce  json
+// @Param id path string true "Employee ID"
+// @Param fullName body string true "fullName" maxlength(99)
+// @Param email body string true "email"
+// @Param phone body string true "phone" minlength(10)  maxlength(12)
+// @Param lineManager body model.UUID true "lineManager"
+// @Param discordID body string true "discordID"
+// @Param githubID body string true "githubID"
+// @Param role body model.UUID true "role"
+// @Param chapter body model.UUID true "chapter"
+// @Param seniority body model.UUID  true "seniority"
+// @Param stack body []model.UUID true "stack"
+// @Param DoB body time.Time true "DoB"
+// @Param gender body string true "gender"
+// @Param address body string true "address" maxlength(200)
+// @Param personalEmail body string true "personalEmail"
+// @Param Authorization header string true "jwt token"
+// @Success 200 {object} view.EditEmployeeResponse
+// @Failure 400 {object} view.ErrorResponse
+// @Failure 404 {object} view.ErrorResponse
+// @Failure 500 {object} view.ErrorResponse
+// @Router /employees/{id}/edit [post]
+func (h *handler) Update(c *gin.Context) {
+	// 1. parse id from uri, validate id
+	var params struct {
+		ID string `uri:"id" binding:"required"`
+	}
+
+	if err := c.ShouldBindUri(&params); err != nil {
+		c.JSON(http.StatusBadRequest, view.CreateResponse[any](nil, nil, err, params))
+		return
+	}
+
+	// 2. parse json body from request
+	type EditEmployeeRequest struct {
+		Fullname    string     `form:"fullName" json:"fullName" binding:"required,max=99"`
+		Email       string     `form:"email" json:"email" binding:"required,email"`
+		Phone       string     `form:"phone" json:"phone" binding:"required,max=12,min=10"`
+		LineManager model.UUID `form:"lineManager" json:"lineManager"`
+		DiscordID   string     `form:"discordID" json:"discordID"`
+		GithubID    string     `form:"githubID" json:"githubID"`
+
+		Role      model.UUID   `form:"role" json:"role" binding:"required"`
+		Chapter   model.UUID   `form:"chapter" json:"chapter"`
+		Seniority model.UUID   `form:"seniority" json:"seniority" binding:"required"`
+		Stack     []model.UUID `form:"stack" json:"stack" binding:"required"`
+
+		DoB           *time.Time `form:"DoB" json:"DoB" binding:"required"`
+		Gender        string     `form:"gender" json:"gender" binding:"required"`
+		Address       string     `form:"address" json:"address" binding:"required,max=200"`
+		PersonalEmail string     `form:"personalEmail" json:"personalEmail" binding:"required,email"`
+	}
+
+	var body EditEmployeeRequest
+
+	if err := c.ShouldBindJSON(&body); err != nil {
+		if err != nil {
+			c.JSON(http.StatusBadRequest, view.CreateResponse[any](nil, nil, err, body))
+			return
+		}
+	}
+
+	// 3. prepare the logger
+	// TODO: can we move this to middleware ?
```

Comment:
remove comment if it's not necessary

Reply:
cái này em không biết sau định udpate không nên em để sẵn cái TODO như bản gốc sau có udpate dễ tìm hết á anh

Reply:
keep TODO logs.